### ISSUE TITLE
Support for Heterogeneous Graphs

### ISSUE THREAD

**Issue by Dsantra92** [CarloLucibello/GraphNeuralNetworks.jl#199]
Now that we support HeteroGeneous Graphs in [MLDatasets.jl](https://github.com/JuliaML/MLDatasets.jl/pull/140). We should support HeteroGeneous graphs.

Ideally,
```julia
g = mldataset2gnngraph(heterogeneous_dataset)
```

**Comment by MilkshakeForReal** [comment]
What about the neural nets?

**Comment by CarloLucibello** [comment]
Once we have heterogeneous graphs, implementing graph convolutional layers will be easy, nailing the design of heterogeneous graphs is what matters. 

For the design, we should get inspiration from PytorchGeometric and DGL, and also be aware that internal dictionaries are a no go since they have multiple [Zygote issues](https://github.com/FluxML/Zygote.jl/issues?q=is%3Aissue+is%3Aopen+label%3Adictionary)

**Comment by Dsantra92** [comment]
This is a duplicate issue, tracking this in #144
